typescript/cli/ci-test.sh

```diff
@@ -2,191 +2,260 @@
-done
-
-echo "Validator running, sleeping to let it sync"
-# This needs to be long to allow time for the cargo build to finish
-sleep 15
-echo "Done sleeping"
-
-echo "Validator Announcement:"
-cat /tmp/anvil1/validator/announcement.json
-
-echo "Pre-building relayer with cargo"
-cargo build --bin relayer
-
-echo "Running relayer"
-export HYP_RELAYCHAINS=anvil1,anvil2
-export HYP_ALLOWLOCALCHECKPOINTSYNCERS=true
-export HYP_DB=/tmp/relayer
-export HYP_GASPAYMENTENFORCEMENT='[{"type":"none"}]'
-export HYP_CHAINS_ANVIL1_SIGNER_TYPE=hexKey
-export HYP_CHAINS_ANVIL1_SIGNER_KEY=0xdbda1821b80551c9d65939329250298aa3472ba22feea921c0cf5d620ea67b97
-export HYP_CHAINS_ANVIL2_SIGNER_TYPE=hexKey
-export HYP_CHAINS_ANVIL2_SIGNER_KEY=0xdbda1821b80551c9d65939329250298aa3472ba22feea921c0cf5d620ea67b97
-export HYP_METRICSPORT=9091
-
-cargo run --bin relayer > /tmp/relayer/relayer-logs.txt &
-
-# This needs to be long to allow time for the cargo build to finish
-sleep 20
-echo "Done running relayer, checking message delivery statuses"
-
-for i in "1 $MESSAGE1_ID" "2 $MESSAGE2_ID"
-do
-    set -- $i
-    echo "Checking delivery status of $1: $2"
-    yarn workspace @hyperlane-xyz/cli run hyperlane status \
-        --id $2 \
+_main() {
+    # the first arg to this script is a flag to enable the hook config as part of core deployment
+    # motivation is to test both the bare bone deployment (included in the docs) and the deployment with the routing over igp hook (which is closer to production deployment)
+    HOOK_FLAG=$1;
+    if [ -z "$HOOK_FLAG" ]; then
+      echo "Usage: fork.sh <hook>";
+      exit 1;
+    fi
+
+    prepare_anvil;
+
+    export DEBUG=hyperlane:*;
+    DEPLOYER=$(cast rpc eth_accounts | jq -r '.[0]');
+
+    run_hyperlane_deploy_core_dry_run;
+
+    reset_anvil;
+
+    run_hyperlane_deploy_core;
+    run_hyperlane_deploy_warp;
+    run_hyperlane_send_message;
+    run_hyperlane_send_transfer;
+
+    cd ./rust;
+
+    run_validator;
+    run_relayer;
+    run_hyperlane_status;
+
+    kill_anvil;
+
+    echo "Done";
+}
+
+prepare_anvil() {
+    # Optional cleanup for previous runs, useful when running locally
+    pkill -f anvil
+    rm -rf /tmp/anvil*
+    rm -rf /tmp/relayer
+
+    if [[ $OSTYPE == 'darwin'* ]]; then
+        # kill child processes on exit, but only locally because
+        # otherwise it causes the script exit code to be non-zero
+        trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT
+    fi
+
+    # Setup directories for anvil chains
+    for CHAIN in anvil1 anvil2
+    do
+        mkdir -p /tmp/$CHAIN /tmp/$CHAIN/state  /tmp/$CHAIN/validator /tmp/relayer
+        chmod -R 777 /tmp/relayer /tmp/$CHAIN
+    done
+
+    anvil --chain-id 31337 -p 8545 --state /tmp/anvil1/state --gas-price 1 > /dev/null &
+    anvil --chain-id 31338 -p 8555 --state /tmp/anvil2/state --gas-price 1 > /dev/null &
+    sleep 1
+
+    set -e
+
+    echo "{}" > /tmp/empty-artifacts.json
```

Comment:
This is more related to the command prep than the anvil prep but nbd

Reply:
ci-test was botched during rebase with `main` so will revisit this